plugins/azure_sentinel/icon_azure_sentinel/util/api.py

```diff
@@ -0,0 +1,275 @@
+import json
+from typing import Any, Dict, List, Tuple, Optional, Union
+import time
+import urllib.parse
+
+import requests
+from icon_azure_sentinel.util.tools import return_non_empty
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+from .endpoints import Endpoint
+
+
+class AzureClient:
+    def __init__(self, logger: Any, tenant_id: str, client_id: str, client_secret: str):
+        self.O365_AUTH_ENDPOINT = "https://login.microsoftonline.com/{}/oauth2/token"
+        self.SCOPE = "https://management.azure.com"
+        self.time_ago = 0  # Jan 1, 1970
+        self.time_now = time.time()  # More than 1 hour since 1978
+        self._auth_token = ""  # nosec
+        self.tenant_id = tenant_id
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.logger = logger
+
+    @property
+    def auth_token(self):
+        tenant_id = self.tenant_id
+        client_id = self.client_id
+        client_secret = self.client_secret
+
+        self.time_now = time.time()
+        if (self.time_now - self.time_ago) < 3500:  # 1 hour in seconds
+            return self._auth_token
```

Comment:
I don't know if it's needed here. I mean it seems that connection part runs everytime any of action starts on the insight platform.